### PR TITLE
WIKI-1473: Wiki Page added from (HOW-TO Guide) Template is not displayed

### PR DESCRIPTION
**Pull request by hassenelkamel-exo**
- The problem was the template is not displayed, so to solve this problem, we implement a new `markupSanitize` function to sanitize perfectly the html .

### CODE REVIEW

File: wiki-service/src/main/java/org/exoplatform/wiki/utils/WikiHTMLSanitizer.java [hunk 1]
```diff
@@ -38,56 +38,34 @@
   private static final Pattern REDO_MACRO_REGEX = Pattern.compile("<wikimacro wikiparam=\"(.*)\"> (.*)</wikimacro>");
 
   public static String markupSanitize(String html) throws Exception {
-    // Replace xwiki comment with specific tags
-    Matcher matcher = LINK_REGEX.matcher(html);
```
Comment (ahamdi) [inline wiki-service/src/main/java/org/exoplatform/wiki/utils/WikiHTMLSanitizer.java]: Please remove all unused variables in this class

File: wiki-service/src/main/java/org/exoplatform/wiki/utils/WikiHTMLSanitizer.java [hunk 2]
```diff
@@ -38,56 +38,34 @@
...
+      Pattern p = Pattern.compile("(<!--.*-->)(.*)");
+      for (int i = 0; i < splittedHTML.length; i++) {
+        if (splittedHTML[i].startsWith("<!--")) {
+          Matcher matcher = p.matcher(splittedHTML[i]);
+          while (matcher.find()) {
+            commentSplitted[i] = matcher.group(1);
+            try {
+              contentSplitted[i] = HTMLSanitizer.sanitize(matcher.group(2));
+            } catch (Exception e) {
+              e.printStackTrace();
```
Comment (ahamdi) [inline wiki-service/src/main/java/org/exoplatform/wiki/utils/WikiHTMLSanitizer.java]: You can remove the try/catch and let the Exception be catched in the upper layer

File: wiki-service/src/test/java/org/exoplatform/wiki/utils/TestWikiHTMLSanitizer.java [hunk 3]
```diff
@@ -10,46 +10,43 @@
 
   public void testWikiEncodeMarkup() {
     /** Encode image attachment **/
-    String html = "<!--startimage:false|-|attach|-|images2.jpg-->" + "<img alt=\"images2.jpg\""
```
Comment (ahamdi) [inline wiki-service/src/test/java/org/exoplatform/wiki/utils/TestWikiHTMLSanitizer.java]: Please add your test case to the other test cases and do not remove them

File: wiki-service/src/main/java/org/exoplatform/wiki/utils/WikiHTMLSanitizer.java (hunk 1 above)
Comment (hassenelkamel-exo) [reply wiki-service/src/main/java/org/exoplatform/wiki/utils/WikiHTMLSanitizer.java]: ok all done!

File: wiki-service/src/main/java/org/exoplatform/wiki/utils/WikiHTMLSanitizer.java (hunk 2 above)
Comment (hassenelkamel-exo) [reply wiki-service/src/main/java/org/exoplatform/wiki/utils/WikiHTMLSanitizer.java]: ok done

File: wiki-service/src/test/java/org/exoplatform/wiki/utils/TestWikiHTMLSanitizer.java (hunk 3 above)
Comment (hassenelkamel-exo) [reply wiki-service/src/test/java/org/exoplatform/wiki/utils/TestWikiHTMLSanitizer.java]: yes done

Comment (ahamdi) [reply wiki-service/src/test/java/org/exoplatform/wiki/utils/TestWikiHTMLSanitizer.java]: Could you restore the first string that contains startimage too.

Comment (hassenelkamel-exo) [reply wiki-service/src/test/java/org/exoplatform/wiki/utils/TestWikiHTMLSanitizer.java]: ok done

File: wiki-service/src/test/java/org/exoplatform/wiki/utils/TestWikiHTMLSanitizer.java [hunk 4]
```diff
@@ -9,25 +9,67 @@
 public class TestWikiHTMLSanitizer extends TestCase {
 
   public void testWikiEncodeMarkup() {
-    /** Encode image attachment **/
-    String html = "<!--startimage:false|-|attach|-|images2.jpg-->" + "<img alt=\"images2.jpg\""
-        + " src=\"/portal/rest/wiki/attachments/draft/space/root/page/root_A_A_557557F072D724C38EA9791A6EDAE263/images2.jpg?width=981\""
-        + " /><!--stopimage-->";
+
+    /** Encode Full macro content **/
```
Comment (ahamdi) [inline wiki-service/src/test/java/org/exoplatform/wiki/utils/TestWikiHTMLSanitizer.java]:
```suggestion
    /** Encode Full Template How-To guide content **/
```

**Pull request merged**